wicket-core/src/main/java/org/apache/wicket/response/LazyStringResponse.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.wicket.response;
+
+import org.apache.wicket.request.Response;
+import org.apache.wicket.util.string.AppendingStringBuffer;
+
+
+/**
+ * Response object that writes to a StringWriter. If the StringResponse is later converted to a
+ * String via toString(), the output which was written to the StringResponse will be returned as a
+ * String.
+ * 
+ * @author Thomas Heigl
+ */
+public class LazyStringResponse extends Response
+{
+
+	/** StringWriter to write to */
+	private AppendingStringBuffer out;
+
+	/**
+	 * @see Response#write(CharSequence)
+	 */
+	@Override
+	public void write(final CharSequence string)
+	{
+		getOut().append(string);
+	}
+
+	/**
+	 * @see Response#reset()
+	 */
+	@Override
+	public void reset()
+	{
+		getOut().clear();
```

Comment:
Is it necessary to create buffer if you only need to reset it?

Maybe this code can be changed to:
`if (out != null) out.reset() ?`

If so `getOut()` method can be dropped :)

Reply:
Ha, i just realized the same thing and pushed it before reading your comment ;)